pkg/source/reconciler/source/kafkasource.go

```diff
@@ -217,6 +217,20 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 			return err
 		}
 	}
+
+	// Propagate deployment status
+	msg, ready, err := r.receiveAdapterStatus(ra)
+	if err != nil {
+		// Unrecoverable error
+		src.Status.MarkNotDeployed("DeploymentError", "%v", err)
+		return nil // Do not retry
+	}
+
+	if !ready {
+		src.Status.MarkNotDeployed("DeploymentNotReady", msg)
+		return errors.New(msg)
```

Comment:
Since we have handlers for the RA deployments [here], when the deployment's status changes we get the associated `KafkaSource` in the queue again, can we use that rather than returning an error for retrying?

Perhaps we can log the message and return `nil`?

[here]: https://github.com/knative-sandbox/eventing-kafka/blob/03ab63a8a07c8e1cdda3148979342f05b4a9e106/pkg/source/reconciler/source/controller.go#L76-L79

Reply:
good point!